test/PowerShellEditorServices.Test.Shared/Refactoring/Functions/ForeachFunction.ps1

```diff
@@ -0,0 +1,17 @@
+$x = 1..10
+
+function testing_files {
+    param (
+        $x
+    )
+    write-host "Printing $x"
+}
+
+foreach ($number in $x) {
+    testing_files $number
+
+    function testing_files {
```

Comment:
Rename on `function` applies to the keyword. I like this behavior if it isn't immediately intuitive, should we do the same for like foreach as well for the inner var?

Reply:
I like this behavior aswell, its a happy byproduct of how the FunctionDefinitionAST works, I don't think it would be too hard to get it working for for / foreach statments. In the screenshot below you can see that the VariableExpressionAst is nicely nested just need some logic to target it when the parent is selected.



![image](https://github.com/PowerShell/PowerShellEditorServices/assets/3089087/05f60732-4064-478a-9caa-8b14d52b956c)